envoyauth/response_test.go

```diff
@@ -0,0 +1,186 @@
+package envoyauth
+
+import (
+	"encoding/json"
+	"testing"
+
+	"google.golang.org/genproto/googleapis/rpc/code"
+)
+
+func TestIsAllowed(t *testing.T) {
+
+	input := make(map[string]interface{})
+	er := EvalResult{
+		Decision: input,
+	}
+	var err error
+
+	_, err = er.IsAllowed()
+	if err == nil {
+		t.Fatal("Expected error but got nil")
+	}
+
+	input["allowed"] = 1
+	_, err = er.IsAllowed()
+	if err == nil {
+		t.Fatal("Expected error but got nil")
+	}
+
+	input["allowed"] = true
+	var result bool
+	result, err = er.IsAllowed()
+
+	if err != nil {
+		t.Fatalf("Expected no error but got %v", err)
+	}
+
+	if result != true {
+		t.Fatalf("Expected result %v but got %v", int32(code.Code_OK), result)
```

Comment:
Does this error message match what we test for? It looks like we're checking if `result` is true or not, and not whether it is `int32(code.Code_OK)`. 🤔

Reply:
Good catch!